### PR TITLE
fix: Prevent metrics from being sent too frequently to OTEL collector

### LINKED ISSUE #250: Duplicate TimeSeries errors from opentelemetry Collector
The following errors are being reported from the OpenTelemetry collector in GKE decoupled mode

```
textPayload: "2024-03-19T09:47:21.215Z	error	exporterhelper/common.go:95	
Exporting failed. Dropping data.	
{
  "kind": "exporter",
  "data_type": "metrics",
  "name": "googlecloud",
  "error": "rpc error: code = InvalidArgument desc = One or more TimeSeries could not be written: Field timeSeries[6] had an invalid value: Duplicate TimeSeries encountered. Only one point can be written per TimeSeries per request.; Field timeSeries[8] had an invalid value: Duplicate TimeSeries encountered. Only one point can be written per TimeSeries per request.; Field timeSeries[7] had an invalid value: Duplicate TimeSeries encountered. Only one point can be written per TimeSeries per request.; Field timeSeries[5] had an invalid value: Duplicate TimeSeries encountered. Only one point can be written per TimeSeries per request.; Field timeSeries[9] had an invalid value: Duplicate TimeSeries encountered. Only one point can be written per TimeSeries per request.\nerror details: name = Unknown  desc = total_point_count:10 success_point_count:5 errors:{status:{code:3} point_count:5}",
  "dropped_items": 10}"
```

### PR DESCRIPTION
**Pull request by nielm**
Fixes #250

When using the GKE decoupled deployment, it is possible that a race condition can occur where the Scaler with a 10s periodic export may report metrics more frequently than the OTEL collector's batching interval (also 10s), leading to multiple sets of metrics being pushed to the downstream metrics manager. This can cause Duplicate TimeSeries errors.

Update the counters to push metrics less frequently in periodic export mode, and ensure that metrics are never flushed manually in this mode.

In addition the OTEL collector in unified mode had a 5s default time for batching which is at the limit of the Google Cloud Monitoring requirements that pushes are at least 5s apart.

### CODE REVIEW

File: kubernetes/decoupled/autoscaler-pkg/poller/poller-hourly.yaml [hunk 1]
```diff
@@ -45,6 +45,8 @@ spec:
                       fieldPath: metadata.name
                 - name: OTLP_COLLECTOR_URL
                   value: "http://otel-collector:4317/"
+                - name: OLTP_IS_LONG_RUNNING_PROCESS
```
Comment (henrybell) [inline kubernetes/decoupled/autoscaler-pkg/poller/poller-hourly.yaml:48]: `OLTP_` -> `OTLP_` throughout. Strictly I suppose `OTEL` would be more accurate here (to refer to OpenTelemetry in general as opposed to its associated protocol) and potentially also in the entry above for consistency, but I don't think this is a show-stopper unless you are keen to update.

File: kubernetes/unified/autoscaler-config/otel-collector.yaml.template [hunk 2]
```diff
@@ -42,6 +48,7 @@ data:
         metrics:
           receivers: [otlp]
           processors: [resourcedetection, batch, memory_limiter]
+          # If using the debug exporter, add it to the following list
```
Comment (henrybell) [inline kubernetes/unified/autoscaler-config/otel-collector.yaml.template:51]: 👍

File: src/autoscaler-common/counters_base.js [hunk 3]
```diff
@@ -73,27 +73,45 @@ const COUNTERS_PREFIX =
...
-  // This is to avoid that the auto-push happens just after a flush
-  // and then raises 'points were written more frequently' errors.
+  // GCM direct pushing is only done in Cloud functions deployments, where
+  // we only flush directly.
   //
-  // However the only Metrics pusher is a periodic one :/
-  [ExporterMode.GCM]: {
-    PERIODIC_FLUSH_INTERVAL: 12 * 60 * 60 * 1_000, // 12hr
+  [ExporterMode.GCM_ONLY_FLUSHING]: {
+    PERIODIC_FLUSH_INTERVAL: 0x7fffffff, // approx 24 days in milliseconds
```
Comment (henrybell) [inline src/autoscaler-common/counters_base.js:86]: To match the comment below, would it help to distinguish between the two modes of operation by using `PERIODIC_EXPORT_INTERVAL` or `BATCH_EXPORT_INTERVAL` or similar to contrast this with the use of "flush" elsewhere?

**Review by henrybell — CHANGES_REQUESTED** [review]

File: kubernetes/decoupled/autoscaler-pkg/poller/poller-hourly.yaml (hunk 1 above)
Comment (nielm) [reply kubernetes/decoupled/autoscaler-pkg/poller/poller-hourly.yaml]: Done

File: src/autoscaler-common/counters_base.js (hunk 3 above)
Comment (nielm) [reply src/autoscaler-common/counters_base.js]: Done

**Review by henrybell — APPROVED** [review]

**Pull request merged**